rules/docker-dhcp-relay.mk

```diff
@@ -17,15 +17,36 @@ $(DOCKER_DHCP_RELAY)_LOAD_DOCKERS = $(DOCKER_CONFIG_ENGINE_BUSTER)
 
 $(DOCKER_DHCP_RELAY)_VERSION = 1.0.0
 $(DOCKER_DHCP_RELAY)_PACKAGE_NAME = dhcp-relay
+$(DOCKER_DHCP_RELAY)_PACKAGE_DEPENDS = database^1.0.0
 
-SONIC_DOCKER_IMAGES += $(DOCKER_DHCP_RELAY)
-SONIC_INSTALL_DOCKER_IMAGES += $(DOCKER_DHCP_RELAY)
+$(DOCKER_DHCP_RELAY)_SERVICE_REQUIRES = updategraph
+$(DOCKER_DHCP_RELAY)_SERVICE_AFTER = swss syncd teamd
+$(DOCKER_DHCP_RELAY)_SERVICE_BEFORE = ntp-config
+$(DOCKER_DHCP_RELAY)_SERVICE_DEPENDENT_OF = swss
 
+SONIC_DOCKER_IMAGES += $(DOCKER_DHCP_RELAY)
 SONIC_DOCKER_DBG_IMAGES += $(DOCKER_DHCP_RELAY_DBG)
-SONIC_INSTALL_DOCKER_DBG_IMAGES += $(DOCKER_DHCP_RELAY_DBG)
+
+ifeq ($(INCLUDE_KUBERNETES),y)
+$(DOCKER_DHCP_RELAY)_DEFAULT_FEATURE_OWNER = kube
+endif
+
+$(DOCKER_DHCP_RELAY)_DEFAULT_FEATURE_STATE_ENABLED = y
+
+ifeq ($(INSTALL_DEBUG_TOOLS),y)
+SONIC_PACKAGES_LOCAL += $(DOCKER_DHCP_RELAY_DBG)
+else
+SONIC_PACKAGES_LOCAL += $(DOCKER_DHCP_RELAY)
+endif
 
 $(DOCKER_DHCP_RELAY)_CONTAINER_NAME = dhcp_relay
-$(DOCKER_DHCP_RELAY)_RUN_OPT += --privileged -t
-$(DOCKER_DHCP_RELAY)_RUN_OPT += -v /etc/sonic:/etc/sonic:ro
-$(DOCKER_DHCP_RELAY)_RUN_OPT += -v /usr/share/sonic/scripts:/usr/share/sonic/scripts:ro
+$(DOCKER_DHCP_RELAY)_CONTAINER_PRIVILEGED = true
+$(DOCKER_DHCP_RELAY)_CONTAINER_VOLUMES += /etc/sonic:/etc/sonic:ro
+$(DOCKER_DHCP_RELAY)_CONTAINER_VOLUMES += /usr/share/sonic/scripts:/usr/share/sonic/scripts:ro
+$(DOCKER_DHCP_RELAY)_CONTAINER_TMPFS += /tmp/
+$(DOCKER_DHCP_RELAY)_CONTAINER_TMPFS += /var/tmp/
+
+$(DOCKER_DHCP_RELAY)_CLI_CONFIG_PLUGIN = /cli/config/plugins/dhcp_relay.py
```

Comment:
how are the cli scripts being exposed to the host os?

Reply:
When sonic-package-manager installs the docker and sees that there are CLI plugins it will install CLI plugins from docker to the host.

Reply:
I see. The only concern I had here is that were exposing files from container to host which in a sense goes against containerization.

Reply:
Yes, you have a valid concern. This was a topic of discussions in SONiC App Ext & Kubernetes subgroup. For phase 1 we defined this simple approach. We aim for CLI autogeneration to solve this.